### PR TITLE
Feature/ml 133 e2e test

### PR DESCRIPTION
**Pull request by PierreLeveau**
**Overview** 
This is the first e2e test.

**Code changes**
* added test data
* Added tests and the data generation script.
* Performed the changes to make the tests pass.

### CODE REVIEW

File: tests/e2e/test_hugging_face_text_classification.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import json
+
+from click.testing import CliRunner
+
+import predict
+import train
+
+content = json.load(open("tests/e2e/fixtures/text_content_fixture.json"))
```
Comment (crsegerie) [inline tests/e2e/test_hugging_face_text_classification.py:8]: rename to text_content

File: tests/utils/gen_hugging_face_text_classification.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    assets = []
+    for i, a in enumerate(g):
+        assets.append(a)
+        if i > 50:
+            break
+
+    for a in assets:
+        for label in a["labels"]:
+            del label["author"]
```
Comment (crsegerie) [inline tests/utils/gen_hugging_face_text_classification.py:22]: This should be in a command  refresh_mock_data.py which would create the mock jsons

File: tests/utils/gen_hugging_face_text_classification.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    with open("tests/e2e/fixtures/text_assets_fixture.json", "w") as f:
+        json.dump(assets, f)
+
+    project = k.projects(project_id="cl1e4umogdgon0ly4737z82lc")[0]  # type:ignore
```
Comment (crsegerie) [inline tests/utils/gen_hugging_face_text_classification.py:27]: k -> Kili

File: tests/e2e/test_hugging_face_text_classification.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+
+def mocked_get_text_from(asset_url):
+    return content[asset_url]
+
+
+def mocked_get_assets(*_, max_assets=None, labeling_statuses=None):
```
Comment (crsegerie) [inline tests/e2e/test_hugging_face_text_classification.py:15]: rename to mocked_text_get_asset

File: tests/e2e/test_hugging_face_text_classification.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+from click.testing import CliRunner
+
+import predict
+import train
+
+content = json.load(open("tests/e2e/fixtures/text_content_fixture.json"))
+
+
+def mocked_get_text_from(asset_url):
```
Comment (crsegerie) [inline tests/e2e/test_hugging_face_text_classification.py:11]: rename to mocked__get_text_from

File: train.py [hunk 6]
```diff
@@ -90,11 +90,11 @@ def train_image_bounding_box(
     default="https://cloud.kili-technology.com/api/label/v2/graphql",
     help="Kili Endpoint",
 )
-@click.option("--api-key", default=os.environ.get("KILI_API_KEY"), help="Kili API Key")
+@click.option("--api-key", default=os.environ.get("KILI_API_KEY", ""), help="Kili API Key")
 @click.option("--model-framework", default=None, help="Model framework (eg. pytorch, tensorflow)")
 @click.option("--model-name", default=None, help="Model name (eg. bert-base-cased)")
 @click.option("--model-repository", default=None, help="Model repository (eg. huggingface)")
-@click.option("--project-id", default=None, help="Kili project ID")
+@click.option("--project-id", default="", help="Kili project ID")
```
Comment (crsegerie) [inline train.py:97]: Shouldn't we delete completely the default ?

Comment (PierreLeveau) [reply train.py:97]: adding `required=True` instead

**Review by crsegerie — CHANGES_REQUESTED** [review]
I left some comments, mostly naming. refresh_mock_data.py is facultative

**Comment by crsegerie** [comment]
lgtm!

**Review by crsegerie — APPROVED** [review]

**Pull request merged**